sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -994,15 +994,17 @@ class SessionCatalog(
     // Go through temporary views and invalidate them.
     // If the database is defined, this may be a global temporary view.
     // If the database is not defined, there is a good chance this is a temp view.
-    if (name.database.isEmpty) {
-      tempViews.get(tableName).foreach(_.refresh())
+    val isTempView = if (name.database.isEmpty) {
```

Comment:
nit: the code structure here looks pretty similar to `isTemporaryTable`, and it would be nice if we can do:
```scala
    lookupTemporaryTable(name).map(_.refresh()).getOrElse {
      // Also invalidate the table relation cache.
      val qualifiedTableName = QualifiedTableName(dbName, tableName)
      tableRelationCache.invalidate(qualifiedTableName)
    }
```

Reply:
I did that.